godot-macros/src/derive/data_models/unit_only_enum.rs

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use proc_macro2::{Ident, Literal, Span, TokenTree};
+
+use crate::util::{bail, error};
+use crate::ParseResult;
+
+/// Stores info from unit-only enums for use in deriving `GodotConvert` and other related traits.
+#[derive(Debug, Clone)]
+pub struct UnitOnlyEnum {
+    /// The names of each variant.
+    names: Vec<Ident>,
+    /// The discriminants of each variant, both explicit and implicit.
+    discriminants: Vec<Literal>,
+}
+
+impl UnitOnlyEnum {
+    /// Parses the enum.
+    ///
+    /// Ensures the enum is unit-only, and that any explicit discriminants are integer literals.
+    pub fn parse_enum(enum_: &venial::Enum) -> ParseResult<Self> {
+        let variants = enum_
+            .variants
+            .items()
+            .map(UnitEnumVariant::parse_enum_variant)
+            .collect::<ParseResult<Vec<_>>>()?;
+
+        let (names, discriminants) = Self::create_discriminant_mapping(variants)?;
+
+        Ok(Self {
+            names,
+            discriminants,
+        })
+    }
+
+    fn create_discriminant_mapping(
+        variants: Vec<UnitEnumVariant>,
+    ) -> ParseResult<(Vec<Ident>, Vec<Literal>)> {
+        // See here for how implicit discriminants are decided
+        // https://doc.rust-lang.org/reference/items/enumerations.html#implicit-discriminants
+        let mut names = Vec::new();
+        let mut discriminants = Vec::new();
+
+        let mut last_discriminant = None;
+        for variant in variants.into_iter() {
+            let discriminant_span = variant.discriminant_span();
+
+            let discriminant = match variant.discriminant_as_i64()? {
+                Some(discriminant) => discriminant,
+                None => last_discriminant.unwrap_or(0) + 1,
+            };
+            last_discriminant = Some(discriminant);
+
+            let mut discriminant = Literal::i64_unsuffixed(discriminant);
+            discriminant.set_span(discriminant_span);
+
+            names.push(variant.name);
+            discriminants.push(discriminant)
+        }
+
+        Ok((names, discriminants))
+    }
```

Comment:
It's probably OK to talk about "discriminants" here, as they refer to the syntactic part at this stage. For example, they can be implicit (`SomeName`) or explicit (`SomeName = 12`).

Calling them "ords" afterwards would then help to separate this syntactic from the semantic concept -- tokens vs. numbers.